Add DiscordSDK.close() to Embedded App Sdk (#6770)
hammerandchisel/discord-api-docs: docs/developer_tools/Embedded_App_SDK.md

| [ready](#DOCS_DEVELOPER_TOOLS_EMBEDDED_APP_SDK/ready)             | Resolves when your app has successfully connected to the Discord client |
| [subscribe](#DOCS_DEVELOPER_TOOLS_EMBEDDED_APP_SDK/subscribe)     | Subscribe to an Embedded App SDK Event                                  |
| [unsubscribe](#DOCS_DEVELOPER_TOOLS_EMBEDDED_APP_SDK/unsubscribe) | Unsubscribe to an Embedded App SDK Event                                |

> | [close](#DOCS_DEVELOPER_TOOLS_EMBEDDED_APP_SDK/close)       | Unsubscribe to an Embedded App SDK Event                                |

Reviewer: ```suggestion
| [close](#DOCS_DEVELOPER_TOOLS_EMBEDDED_APP_SDK/close)             | Close an Embedded App                                                   |
```

Author: Maybe

---

### close()

> Used to close your app with specific code and reason

Reviewer: ```suggestion
Used to close your app with a specified code and reason.
```

#### Supported Platforms
| Web | iOS | Android |
|-----|-----|---------|
| ✅   | ✅   | ✅       |

#### Required Scopes

No scopes required

#### SDK Usage

```js

> async function setup() {
>   await discordSdk.ready();
>   discordSdk.close(RPCCloseCodes.CLOSE_NORMAL, "You exited from app");
> }

Reviewer: ```suggestion
discordSdk.close(RPCCloseCodes.CLOSE_NORMAL, "You exited from app");
```

Is close `async`? Not sure right now.
Otherwise
```js
await discordSdk.close(RPCCloseCodes.CLOSE_NORMAL, "You exited from app");
```

Reviewer 2: > Is close `async`? Not sure right now. Otherwise
> 
> ```js
> await discordSdk.close(RPCCloseCodes.CLOSE_NORMAL, "You exited from app");
> ```

[no](https://github.com/discord/embedded-app-sdk/blob/main/src/Discord.ts#L144-L149)

---

### close()

Used to close your app with a specified code and reason.

#### Supported Platforms
| Web | iOS | Android |
|-----|-----|---------|
| ✅   | ✅   | ✅       |

#### Required Scopes

No scopes required

#### RPC Close Codes

> | Name | Code |
> |-----|-----|
> | CLOSE_NORMAL | 1000 |
> | CLOSE_UNSUPPORTED | 1003 |
> | CLOSE_ABNORMAL | 1006 |
> | INVALID_CLIENTID | 4000 |
> | INVALID_ORIGIN | 4001 |
> | RATELIMITED | 4002 |
> | TOKEN_REVOKED | 4003 |
> | INVALID_VERSION | 4004 |
> | INVALID_ENCODING | 4005 |

Reviewer: ```suggestion
| Name              | Code |
|-------------------|------|
| CLOSE_NORMAL      | 1000 |
| CLOSE_UNSUPPORTED | 1003 |
| CLOSE_ABNORMAL    | 1006 |
| INVALID_CLIENTID  | 4000 |
| INVALID_ORIGIN    | 4001 |
| RATELIMITED       | 4002 |
| TOKEN_REVOKED     | 4003 |
| INVALID_VERSION   | 4004 |
| INVALID_ENCODING  | 4005 |
```

Table formatting..
